presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -135,6 +135,42 @@ public static Slice stAsText(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
         return utf8Slice(deserialize(input).asText());
     }
 
+    @Description("Returns the geometry type of the geometry")
+    @ScalarFunction("ST_GeometryType")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice stGeometryType(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
+    {
+        GeometryType geometryType = deserializeType(input);
+
+        String type;
+        switch (geometryType) {
+            case POINT:
+                type = "ST_Point";
+                break;
+            case MULTI_POINT:
+                type = "ST_MultiPoint";
+                break;
+            case LINE_STRING:
+                type = "ST_LineString";
+                break;
+            case MULTI_LINE_STRING:
+                type = "ST_MultiLineString";
+                break;
+            case POLYGON:
+                type = "ST_Polygon";
+                break;
+            case MULTI_POLYGON:
+                type = "ST_MultiPolygon";
+                break;
+            case GEOMETRY_COLLECTION:
+                type = "ST_GeomCollection";
+                break;
+            default:
+                throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Unexpected geometry type: " + geometryType);
```

Comment:
What about ENVELOPE?

assuming it's possible that someone adds new GeometryType someday, could we make sure that this code gets updated too?

Maybe you could have a test that:
1. constructs a map `GeometryType → (geometry text, expected output from ST_GeometryType)`
2. iterates over `GeometryType.values()` and runs a test based on entries from the map
3. any geo type not found in this map causes test failure

this way we are sure all geo types are exercised in tests with this function

Reply:
Done. Updated below.

I set the result as "ST_Envelope" for envelopes, although I'm not totally sure that's standard.

Reply:
@jwass @findepi @arhimondr GeometryType enum lists types used for serialization. Envelope type is used to efficiently serialize trivial polygons and enable further optimizations by preserving the fact that the polygon is a rectangle aligned with coordinate axes across ser-de (see #10327).

Andrii pointed out that it would be clearer to rename GeometryType to GeometrySerializationType . If we do that, we could then add `getStandardName` method to the enum to return the type name that ST_GeometryType expects. This way when new serialization type is added, it will be obvious that standard name needs to be provided for it.

ST_GeometryType must return ST_Polygon for envelopes.

Reply:
Yes. I would rename `GeometryType` to `GeometrySerializationType`. Than you can introduce new enum called `GeometryType` that contains all the fields from `GeometrySerializationType`, but `Envelope`. Than in `GeometrySerializationType` introduce field typed `GeometryType` and set it to `POLYGON` for both, `POLYGON` and `ENVELOPE`. Also move `multitype` flag from `GeometrySerializationType` to `GeometryType`.